SonarQube.Common/VerbosityCalculator.cs

```diff
@@ -0,0 +1,109 @@
+{
+    public static class VerbosityCalculator
+    {
+        /// <summary>
+        /// The initial verbosity of loggers
+        /// </summary>
+        /// <remarks>
+        /// Each of the executables (bootstrapper, pre/post processor) have to parse their command line and config files before
+        /// being able to deduce the correct verbosity. In doing that it makes sense to set the logger to be more verbose before
+        /// a value can be set so as not to miss out on messages.
+        /// </remarks>
+        public const LoggerVerbosity InitialLoggingVerbosity = LoggerVerbosity.Debug;
+
+        /// <summary>
+        /// The verbosity of loggers if no settings have been specified
+        /// </summary>
+        public const LoggerVerbosity DefaultLoggingVerbosity = LoggerVerbosity.Info;
+
+        /// <summary>
+        /// Computes verbosity based on the available properties. 
+        /// </summary>
+        /// <remarks>If no verbosity setting is present, the default verbosity (info) is used</remarks>
+        public static LoggerVerbosity ComputeVerbosity(IAnalysisPropertyProvider properties, ILogger logger)
+        {
+            if (properties == null)
+            {
+                throw new ArgumentNullException("properties");
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            string sonarVerboseValue;
+            properties.TryGetValue(SonarProperties.Verbose, out sonarVerboseValue);
+
+            string sonarLogLevelValue;
+            properties.TryGetValue(SonarProperties.LogLevel, out sonarLogLevelValue);
+
+            return ComputeVerbosity(sonarVerboseValue, sonarLogLevelValue, logger);
+        }
+
+        /// <summary>
+        /// Computes verbosity based on the analysis config file settings. 
+        /// </summary>
+        /// <remarks>If no verbosity setting is present, the default verbosity (info) is used</remarks>
+        public static LoggerVerbosity ComputeVerbosity(AnalysisConfig config, ILogger logger)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            AnalysisSetting sonarVerboseSetting;
+            config.TryGetSetting(SonarProperties.Verbose, out sonarVerboseSetting);
+
+            AnalysisSetting sonarLogLevelSetting;
+            config.TryGetSetting(SonarProperties.LogLevel, out sonarLogLevelSetting);
+
+            return ComputeVerbosity(
+                sonarVerboseSetting == null ? null : sonarVerboseSetting.Value,
+                sonarLogLevelSetting == null ? null : sonarLogLevelSetting.Value,
+                logger);
+        }
+
+        private static LoggerVerbosity ComputeVerbosity(string sonarVerboseValue, string sonarLogValue, ILogger logger)
```

Comment:
sonar.verbose has the priority over sonar.log.level - I will add this to the method comments